### PR TITLE
tweak(platform): Update block descriptions and tweak block card ui

### PR DESCRIPTION
**Pull request by Swiftyos**
### Background

Changed blocks to be added when clicked, tweaked block cards in block list 

### Changes 🏗️
 
- add when clicking anywhere in the block

Screenshot 2024-09-24 at 12 18 40

Only for the new autogpt platform, currently in autogpt_platform/

- Create from scratch and execute an agent with at least 3 blocks
- Import an agent from file upload, and confirm it executes correctly
- Upload agent to marketplace
- Import an agent from marketplace and confirm it executes correctly
- Edit an agent from monitor, and confirm it executes correctly

### CODE REVIEW

**Comment by Swiftyos** [comment]
@Significant-Gravitas/maintainers Given we are so close to release. Please can you be extra diligent during reviews to ensure no bugs slip through

File: autogpt_platform/frontend/src/components/edit/control/BlocksControl.tsx [hunk 1]
```diff
@@ -132,33 +132,31 @@ export const BlocksControl: React.FC<BlocksControlProps> = ({
               {filteredBlocks.map((block) => (
                 <Card
                   key={block.id}
-                  className={`m-2 ${getPrimaryCategoryColor(block.categories)}`}
+                  className={`m-2 my-4 flex h-20 border`}
```
Comment (ntindle) [inline autogpt_platform/frontend/src/components/edit/control/BlocksControl.tsx]:
```suggestion
                  className="m-2 my-4 flex h-20 border"
```

File: autogpt_platform/backend/backend/blocks/basic.py [hunk 2]
```diff
@@ -41,8 +41,7 @@ class Output(BlockSchema):
     def __init__(self):
         super().__init__(
             id="1ff065e9-88e8-4358-9d82-8dc91f622ba9",
-            description="This block forwards the `input` pin to `output` pin. "
-            "This block output will be static, the output can be consumed many times.",
+            description="Forwards the `input` to `output`, providing static output that can be reused.",
```
Comment (ntindle) [inline autogpt_platform/backend/backend/blocks/basic.py]: This still makes very little sense to a normal person 

try replacing static with something like kept output?

File: autogpt_platform/backend/backend/blocks/basic.py [hunk 3]
```diff
@@ -245,12 +244,7 @@ def __init__(self):
         super().__init__(
             id="363ae599-353e-4804-937e-b2ee3cef3da4",
             description=(
-                "This block records the graph output. It takes a value to record, "
-                "with a name, description, and optional format string. If a format "
-                "string is given, it tries to format the recorded value. The "
-                "formatted (or raw, if formatting fails) value is then output. "
-                "This block is key for capturing and presenting final results or "
-                "important intermediate outputs of the graph execution."
+                "Records the graph output to be displayed in the output panel."
```
Comment (ntindle) [inline autogpt_platform/backend/backend/blocks/basic.py]: not sure what this means too

File: autogpt_platform/backend/backend/blocks/csv.py [hunk 4]
```diff
@@ -22,6 +22,7 @@ def __init__(self):
             id="acf7625e-d2cb-4941-bfeb-2819fc6fc015",
             input_schema=ReadCsvBlock.Input,
             output_schema=ReadCsvBlock.Output,
+            description="Reads a CSV file and outputs the data as a list of dictionaries.",
```
Comment (ntindle) [inline autogpt_platform/backend/backend/blocks/csv.py]: and individual rows via rows

**Review by ntindle — CHANGES_REQUESTED** [review]

File: autogpt_platform/backend/backend/blocks/basic.py (hunk 2 above)
Comment (Swiftyos) [reply autogpt_platform/backend/backend/blocks/basic.py]: Updated

File: autogpt_platform/backend/backend/blocks/csv.py (hunk 4 above)
Comment (Swiftyos) [reply autogpt_platform/backend/backend/blocks/csv.py]: updated

File: autogpt_platform/backend/backend/blocks/basic.py [hunk 5]
```diff
@@ -41,8 +41,7 @@ class Output(BlockSchema):
     def __init__(self):
         super().__init__(
             id="1ff065e9-88e8-4358-9d82-8dc91f622ba9",
-            description="This block forwards the `input` pin to `output` pin. "
-            "This block output will be static, the output can be consumed many times.",
+            description="This block forwards an input value as output, allowing reuse without change.",
```
Comment (majdyz) [inline autogpt_platform/backend/backend/blocks/basic.py:44]: The static information is really important but Idk what is the better way to convey it.
This is a special block when you use it, the link connected to other nodes will look and behave differently:

image

The output produced by this block can be reused and not gone when its consumed.

File: autogpt_platform/backend/backend/blocks/discord.py [hunk 6]
```diff
@@ -31,6 +31,7 @@ def __init__(self):
             id="d3f4g5h6-1i2j-3k4l-5m6n-7o8p9q0r1s2t",  # Unique ID for the node
             input_schema=ReadDiscordMessagesBlock.Input,  # Assign input schema
             output_schema=ReadDiscordMessagesBlock.Output,  # Assign output schema
+            description="Reads messages from a Discord channel using a bot token.",
```
Comment (majdyz) [inline autogpt_platform/backend/backend/blocks/discord.py:34]: is the bot token part required?

**Review by majdyz — APPROVED** [review]

File: autogpt_platform/backend/backend/blocks/discord.py (hunk 6 above)
Comment (ntindle) [reply autogpt_platform/backend/backend/blocks/discord.py:34]: aside: discord blocks should be updated to the credential system

Comment (ntindle) [reply autogpt_platform/backend/backend/blocks/discord.py:34]: ticket created
https://github.com/Significant-Gravitas/AutoGPT/issues/8170

**Pull request merged**